s2a_options.go

```diff
@@ -121,6 +125,34 @@ type ClientOptions struct {
 	// VerificationMode specifies the mode that S2A must use to verify the
 	// peer certificate chain.
 	VerificationMode VerificationModeType
+
+	// S2Av2 only.
+	// Optional fallback after dialing with S2Av2 fails.
+	FallbackOpts *FallbackOptions
+}
+
+// FallbackOptions contains the fallback options for S2Av2.
+type FallbackOptions struct {
+	// FallbackClientHandshakeFunc is used to specify fallback behavior when calling s2a.NewClientCreds().
+	// It will be called by s2av2TransportCreds's ClientHandshake function, after handshake with S2Av2 fails.
+	//     originServer: the original server attempted with S2Av2.
+	//     originConn: the original raw tcp connection passed into S2Av2's ClientHandshake func.
+	//                 If fallback is successful, the `originConn` should be closed.
+	//     originErr: the error encountered when performing client handshake with S2Av2.
+	// This fallback func should return a post-handshake TLS connection, plus its auth info.
```

Comment:
nit: Please remove "post-handshake". From the user perspective, this should just be a TLS connection and they should not need to reason about the protocol details.